quinn-proto/src/packet.rs

```diff
@@ -138,7 +138,7 @@ impl PartialDecode {
             invariant_header,
             mut buf,
         } = self;
-        let (payload_len, header) = match invariant_header {
+        Ok(match invariant_header {
```

Comment:
So if I understand correctly, this more or less duplicates some code 5 times in order to special case one of the occurrences? It feels like there might be a better way to do so, maybe by adding another value to the intermediate tuple that is `Some((token_pos, token_len))` for the start and end of the token?

Reply:
The refactoring I did for the above issue makes this a bit cleaner too, I think.